nengo/networks/linear_system.py

```diff
@@ -0,0 +1,216 @@
+    an accurate implementation of the system using the provided synapse.
+
+    Parameters
+    ----------
+    system : `.LinearSystem`
+        The LinearSystem to be implemented.
+    synapse : `.LinearFilter`
+        The synapse to use on the recurrent connection and input connection. Must be
+        an instance or subclass of `.LinearFilter`.
+    n_neurons : int
+        The number of neurons per ensemble in the state `.EnsembleArray`.
+    dt : float (optional)
+        The time constant used to discretize the synapses. If provided, it should be
+        the same as the simulator time constant. If not provided (default), the system
+        will not account for the effects of discretization.
+
+    Attributes
+    ----------
+    mapped_system : `.LinearSystem`
+        The mapped linear system that will be implemented by the network. It accounts
+        for the effect of the recurrent/input synapses on the dynamics.
+    input : `.Node`
+        A passthrough `.Node` for connecting the system input.
+    state : `.EnsembleArray`
+        The object representing the system state.
+    state_input : `.Node`
+        The input to the system state.
+    state_output : `.Node`
+        The output from the system state.
+    output : `.Node`
+        A passthrough `.Node` for retrieving the system output.
+    ss_connections : dict of (str, `.Connection`)
+        Dictionary containing a `.Connection` for each of the state-space transform
+        matrices "A", "B", "C", and "D". "B" will not be present if the system does
+        not have input, and "D" will not be present if the ``D`` matrix is zero.
+
+    See Also
+    --------
+    :func:`.ss2sim`
+    """
+
+    system = LinearSystemParam("system")
+    synapse = LinearFilterParam("synapse")
+
+    def __init__(
+        self,
+        system,
+        synapse,
+        n_neurons=100,
+        dt=None,
+        output_synapse=None,
+        label=None,
+        seed=None,
+        add_to_container=None,
+        **ens_kwargs
+    ):
+        super().__init__(label, seed, add_to_container)
+
+        self.system = system
+        self.synapse = synapse
+        self.mapped_system = ss2sim(self.system, self.synapse, dt=dt)
+        A, B, C, D = self.mapped_system.ss
+
+        with self:
+            # --- make state ensemble
+            x_dims = self.mapped_system.state_size
+            x_label = "%s:state" % ("" if self.label is None else self.label)
+            ens_kwargs.setdefault("label", x_label)
+            self.state, self.state_input, self.state_output = self.make_state_object(
+                n_neurons, x_dims, **ens_kwargs
+            )
+
+            # --- make connections
+            self.ss_connections = {}
+            self.ss_connections["A"] = nengo.Connection(
+                self.state_output,
+                self.state_input,
+                transform=A,
+                synapse=self.synapse.copy(initial_output=self.mapped_system.x0),
```

Comment:
What would happen if the synapse they pass in already has an `initial_output` specified (in relation to what it would do here and below on the "B" connection)?

Reply:
Ahh, yeah, good point. Right now, it would be a back-door undocumented way to set the initial input to the network, except it's setting that initial input *after* the `B` transform, which is weird. Probably best to make that synapse a copy, too, with an initial output of 0.

Reply:
Maybe could throw an error if they pass in a synapse with an initial output, telling the user to set it on the system instead? That way it's clear that the case is not currently supported (and not really well-defined either). I think what it would end up doing right now is (indirectly) add the extra initial value to the same initial state `x0`. As you say I don't think this would be used much, so this would just be a safeguard for a user doing something weird (and probably not what they want to be doing).

Reply:
Yeah, I think an error is a good idea. I've added one. It's possible that it might be annoying occasionally (e.g. if the user's already got a synapse with an `initial_output=Choice([0])`, they'd get the error, even though that's identical to `initial_output=None`), but I think with the new ability to copy a FrozenObject and change a value at the same time, it should be easy to work around (i.e. passing in `synapse.copy(initial_output=None)` would always get around the error).